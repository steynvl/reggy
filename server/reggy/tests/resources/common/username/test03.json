{
  "type": "Username",
  "information": {
    "shouldStartWith": "Letter",
    "shouldContain": [
      "Lowercase letter",
      "Special character",
      "Number",
      "Uppercase letter"
    ],
    "minimumLength": "23",
    "maximumLength": "55"
  },
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "commonUseCases"
}