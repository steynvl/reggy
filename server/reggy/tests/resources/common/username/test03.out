Java
Pattern regex = Pattern.compile("[A-Za-z](?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*\\d)(?=.*[A-Z]).{22,54}");
[A-Za-z](?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*\\d)(?=.*[A-Z]).{22,54}
Perl
my $regex = /[A-Za-z](?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}/;
[A-Za-z](?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}
POSIX
[A-Za-z](?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}
[A-Za-z](?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}
Python
regex = re.compile(r'[A-Za-z](?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}')
[A-Za-z](?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}
JavaScript
const regex = /[A-Za-z](?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}/;
[A-Za-z](?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}
PHP
$regex = '/[A-Za-z](?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}/';
[A-Za-z](?=.*[a-z])(?=.*[!"#$%&'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}
Golang
regex, _ := regexp.Compile("[A-Za-z](?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*\\d)(?=.*[A-Z]).{22,54}")
[A-Za-z](?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*\\d)(?=.*[A-Z]).{22,54}
Rust
let re = Regex::new(r"[A-Za-z](?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}").unwrap();
[A-Za-z](?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}
C#
var re = new Regex(@"[A-Za-z](?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}");
[A-Za-z](?=.*[a-z])(?=.*[!"#$%&\'()*+,-.:;<=>?@\[\]^_`{|}~])(?=.*\d)(?=.*[A-Z]).{22,54}
Scala
val re = "[A-Za-z](?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*\\d)(?=.*[A-Z]).{22,54}".r
[A-Za-z](?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*\\d)(?=.*[A-Z]).{22,54}
Kotlin
val regex = Regex("[A-Za-z](?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*\\d)(?=.*[A-Z]).{22,54}")
[A-Za-z](?=.*[a-z])(?=.*[!\"#$%&'()*+,-.:;<=>?@\\[\\]^_`{|}~])(?=.*\\d)(?=.*[A-Z]).{22,54}